### ISSUE TITLE
CPU_NUM 设置为10训练一个batch_size = 500的样本报错

### ISSUE THREAD

**Issue by TomorrowIsAnOtherDay** [PaddlePaddle/Paddle#18809]
![image](https://user-images.githubusercontent.com/25046619/61854639-1984dd80-aef1-11e9-82be-0aee2829a47c.png)
使用PE的CPU版本进行训练，通过环境变量设置：`export CPU_NUM=10`，训练一个batch_size为500的batch时报错。
如果设置`CPU_NUM = 1`则不会出现问题。
报错信息如上，使用paddle版本为1.5.1

**Comment by wanghaoshuang** [comment]
你好，麻烦贴下代码？log里提示batch size是1. 训练时，feed的数据真的是batch size为500么？可以通过fluid.layers.Print(intput, summarize=10, message="input")看一下。

**Comment by TomorrowIsAnOtherDay** [comment]
感谢答复！
我这边发现了问题所在：
在训练的时候我需要传入一个超参数给模型（类似学习率这种性质的参数）
所以我在feed每个batch的时候需要传入一个数值，这就导致了检查不通过呢。

```python
        total_loss, pi_loss, vf_loss, entropy = self.fluid_executor.run(
            feed={
                'obs': obs_np,
                'actions': actions_np,
                'advantages': advantages_np,
                'target_values': target_values_np,
                'lr': np.array([lr], dtype='float32'),
                'entropy_coeff': np.array([entropy_coeff], dtype='float32')
            },
            fetch_list=self.learn_outputs)
```
如果我将后面的两个超参数去掉的话，则可以正常训练。
在这种情况下有什么建议吗？我既想传入超参数，也想使用PE。

**Comment by wanghaoshuang** [comment]
你要传入的超参来自哪儿呢？数据文件？根据当前batch id计算出来的？

- 数据文件： 把超参repeat成[batch_size]
- 根据当前batch计算：可以将超参放入一个Variable中，具体可参考：[lr scheduler](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/learning_rate_scheduler.py)的实现

**Comment by TomorrowIsAnOtherDay** [comment]
了解了，谢谢解答。

**Comment by chengduoZH** [comment]
@TomorrowIsAnOtherDay 可以贴一下你的模型吗？从报错信息上有点奇怪，你上面介绍已经设置了`export CPU_NUM=10`，但是现实的错误还是CPU_NUM为1.

**Comment by TomorrowIsAnOtherDay** [comment]
实际的使用代码是这里的，我们目前为了避免这个问题，限定了CPU_NUM=1：
https://github.com/PaddlePaddle/PARL/tree/develop/examples/A2C
你可以先尝试用这份代码跑一下看下能否复现问题。
如果没法复现的话请联系我，我会重新写一份更精简的代码来描述当前问题。

**Comment by chengduoZH** [comment]
收到，谢谢
